manifests/repo.pp

```diff
@@ -0,0 +1,17 @@
+class katello::repo (
+  $manage_repo = $::katello::manage_repo,
+  $repo        = $::katello::repo,
+  $dist        = $::katello::repo_yumcode,
+  $gpgcheck    = $::katello::repo_gpgcheck,
+  $gpgkey      = $::katello::repo_gpgkey,
+) {
+  if $manage_repo {
```

Comment:
Why not move that if to init.pp?

if $manage_repo {
 include ::katello::repo
}